mptt/models.py

```diff
@@ -910,15 +910,14 @@ def save(self, *args, **kwargs):
                 same_order = True
 
             if not same_order:
+                parent = getattr(self, opts.parent_attr)
```

Comment:
I assume this could cause a query, hence skipping it if `opts.order_insertion_by or parent_id is not None` is falsey might be an optimisation, but I'm not sure it's worth it.